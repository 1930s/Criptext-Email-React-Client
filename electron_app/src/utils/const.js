const PROD_SOCKET_URL = 'wss://socket.criptext.com';
const PROD_SERVER_URL = 'https://api.criptext.com';
const FILE_SERVER_APP_ID = 'qynhtyzjrshazxqarkpy';
const FILE_SERVER_KEY = 'lofjksedbxuucdjjpnby';

const appDomain = 'criptext.com';

module.exports = {
  FILE_SERVER_APP_ID,
  FILE_SERVER_KEY,
  PROD_SERVER_URL,
  PROD_SOCKET_URL,
  appDomain
};
