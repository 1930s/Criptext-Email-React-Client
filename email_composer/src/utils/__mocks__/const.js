export const appDomain = 'criptext.com';

export const composerEvents = {
  EDIT_DRAFT: 'edit-draft',
  FORWARD: 'forward',
  REPLY: 'reply',
  REPLY_ALL: 'reply-all',
  NEW_WITH_DATA: 'new-with-data'
};
