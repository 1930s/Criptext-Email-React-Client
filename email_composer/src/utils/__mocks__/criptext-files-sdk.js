export default class FileManager {
  constructor() {
    const Event = {
      FILE_PROGRESS: 'progress',
      FILE_FINISH: 'finish',
      FILE_ERROR: 'error'
    };
    return { Event };
  }
}
