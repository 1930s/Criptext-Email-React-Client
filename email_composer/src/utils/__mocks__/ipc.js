import file from './../../../public/emails.json';
const emails = file.emails;

export const getEmailByKey = key => {
  return emails.filter(email => email.key === key);
};
