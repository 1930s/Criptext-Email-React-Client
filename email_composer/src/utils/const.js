/*global process */

export const appDomain =
  process.env.NODE_ENV === 'development' ? 'jigl.com' : 'criptext.com';
