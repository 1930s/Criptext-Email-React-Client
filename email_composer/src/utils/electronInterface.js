const electron = window.require('electron');
const remote = electron.remote;
const dbManager = remote.require('./src/DBManager');

export const createSignalstore = params => {
  return dbManager.createSignalstore(params);
};

export const createKeys = params => {
  return dbManager.createKeys(params);
};

export const getKeys = params => {
  return dbManager.getKeys(params);
};

export const getIdentityKeyPair = params => {
  return dbManager.getIdentityKeyPair(params);
};

export const getRegistrationId = params => {
  return dbManager.getRegistrationId(params);
};

export const getPreKeyPair = params => {
  return dbManager.getPreKeyPair(params);
};

export const getSignedPreKey = params => {
  return dbManager.getSignedPreKey(params);
};

export const createSession = params => {
  return dbManager.createSession(params);
};

export const getKeyserverToken = () => {
  return dbManager.getKeyserverToken();
};
