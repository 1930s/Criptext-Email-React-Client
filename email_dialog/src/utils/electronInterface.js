const electron = window.require('electron');
const { ipcRenderer, remote } = electron;
const globalManager = remote.require('./src/globalManager');

export const remoteData = globalManager.modalData.get();

/* Window events
   ----------------------------- */
export const onResponseModal = (event, response, sendTo) => {
  ipcRenderer.send('response-modal', response, sendTo);
};
