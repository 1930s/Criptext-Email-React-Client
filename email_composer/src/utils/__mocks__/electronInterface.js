import { appDomain } from './../const';

export const myAccount = {
  recipientId: 'julian'
};

const emails = [
  {
    id: 1,
    key: '1',
    threadId: '1',
    s3Key: '1',
    content: 'Email content 1',
    preview: 'Email preview 1',
    subject: 'Email subject 1',
    date: 1514231264772,
    delivered: 0,
    unread: 1,
    secure: 1,
    isMuted: 0
  },
  {
    id: 2,
    key: '2',
    threadId: '2',
    s3Key: '2',
    content: 'Email content 2',
    preview: 'Email preview 2',
    subject: 'Email subject 2',
    date: 1515624279364,
    delivered: 0,
    unread: 1,
    secure: 1,
    isMuted: 0
  },
  {
    id: 3,
    key: '3',
    threadId: '3',
    s3Key: '3',
    content: 'Email content 3',
    preview: 'Email preview 3',
    subject: 'Email subject 3',
    date: 1512744868344,
    delivered: 0,
    unread: 1,
    secure: 1,
    isMuted: 0
  }
];
const contacts = [
  {
    id: 1,
    name: 'Usuario 1',
    email: `toUser@${appDomain}`
  },
  {
    id: 2,
    name: 'Usuario 2',
    email: `ccUser@${appDomain}`
  },
  {
    id: 3,
    name: 'Usuario 3',
    email: `bccUser@${appDomain}`
  },
  {
    id: 4,
    name: 'Usuario 4',
    email: `fromUser@${appDomain}`
  }
];

export const emailId = 1;
export const emailKey = '1';

export const getEmailByKey = key => {
  return emails.filter(email => email.key === key);
};

export const getContactsByEmailId = id => {
  const response = {
    to: [{ email: contacts[0].email }],
    cc: [{ email: contacts[1].email }],
    bcc: [{ email: contacts[2].email }],
    from: [{ email: contacts[3].email }]
  };
  const emptyResponse = {
    to: [],
    cc: [],
    bcc: [],
    from: []
  };
  return id === emailId ? response : emptyResponse;
};
