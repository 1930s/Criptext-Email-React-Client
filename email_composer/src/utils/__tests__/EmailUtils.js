/* eslint-env node, jest */

import { EditorState } from 'draft-js';
import {
  formOutgoingEmailFromData,
  formDataToFillComposer
} from './../EmailUtils';
import { emailKey } from './../__mocks__/electronInterface';
import { appDomain } from './../const';

jest.mock('./../../utils/electronInterface');

describe('Email utils:', () => {
  it('Form outgoing email from data', () => {
    const labelId = 6;
    const composerData = {
      toEmails: [`toUser@${appDomain}`],
      ccEmails: [`ccUser@${appDomain}`],
      bccEmails: [`bccUser@${appDomain}`],
      textSubject: 'Subject',
      htmlBody: EditorState.createEmpty()
    };

    const outgoingData = formOutgoingEmailFromData(composerData, labelId);
    outgoingData.data.email.date = 1524861748481;
    outgoingData.data.email.key = 1524861748481;
    expect(outgoingData).toMatchSnapshot();
  });

  it('Form data to edit draft', async () => {
    const dataToEdit = await formDataToFillComposer(emailKey);
    dataToEdit.htmlBody = '<p>Modified body content</p>';
    expect(dataToEdit).toMatchSnapshot();
  });
});
