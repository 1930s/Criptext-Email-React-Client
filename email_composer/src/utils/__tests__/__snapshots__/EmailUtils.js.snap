// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email utils: Form data to edit draft 1`] = `
Object {
  "bccEmails": Array [
    "bccUser@jigl.com",
  ],
  "ccEmails": Array [
    "ccUser@jigl.com",
  ],
  "htmlBody": "<p>Modified body content</p>",
  "textSubject": "Email subject 1",
  "toEmails": Array [
    "toUser@jigl.com",
  ],
}
`;

exports[`Email utils: Form outgoing email from data 1`] = `
Object {
  "body": "<p></p>
",
  "data": Object {
    "email": Object {
      "content": "<p></p>
",
      "date": 1524861748481,
      "delivered": false,
      "isMuted": false,
      "key": 1524861748481,
      "preview": "
",
      "secure": true,
      "subject": "Subject",
      "unread": false,
    },
    "labels": Array [
      6,
    ],
    "recipients": Object {
      "bcc": Array [
        "bccUser@jigl.com",
      ],
      "cc": Array [
        "ccUser@jigl.com",
      ],
      "from": Array [
        "julian@jigl.com",
      ],
      "to": Array [
        "toUser@jigl.com",
      ],
    },
  },
  "subject": "Subject",
  "to": Array [
    Object {
      "recipientId": "toUser",
      "type": "to",
    },
    Object {
      "recipientId": "ccUser",
      "type": "cc",
    },
    Object {
      "recipientId": "bccUser",
      "type": "bcc",
    },
  ],
}
`;
